frontend/src/main/java/org/abs_models/frontend/typechecker/locationtypes/Constraints.java

```diff
@@ -0,0 +1,92 @@
+package org.abs_models.frontend.typechecker.locationtypes;
```

Comment:
This class maybe wants to be renamed?  There's `Constraint` and `Constraints` with only one letter difference, and no documentation in either.  In general, a file / class header that says something like "This class is part of the XXX system.  Its purpose is XXX, which it does together with YYY.  The main entry points are ZZZ." would be appreciated.

Reply:
That probably is confusing, and an explanation makes a lot of sense. I will do that.

Reply:
Many thanks! In general, some short bits of documentation for added classes and aspects help a lot.  Didn't bother to write this at all points, just where it jumped out at me most.